File: sort/selection_sort.rs
// Selection sort

fn main() {
    let v = vec![8, 4, 9, 12, 1, 47, 3, -6, 15, 0];
    println!("{:?}", v);
    let sorted = selection_sort(&v);
    println!("{:?} — sorted", sorted);
}

fn selection_sort(v: &Vec<i32>) -> Vec<i32> {
    let mut sorted = vec![];
    let mut unsorted = v.clone();
    while unsorted.len() > 0 {
        let mut min_index = 0;
        for (i, x) in unsorted.clone().into_iter().enumerate() {
            if x < unsorted[min_index] {
                min_index = i;
            }
        }
        sorted.push(unsorted.remove(min_index));
    }
    return sorted;
}

File: sort/selection_sort.py
# Selection sort

def sort(arr):
    unsorted = arr
    sorted = []
    while unsorted:
        index_of_min = 0
        min = arr[0]
        for i, x in enumerate(unsorted):
            if x < min:
                min = x
                index_of_min = i
        sorted.append(unsorted.pop(index_of_min))
    return sorted

# Test
arr = [3, 5, 13, 7, 0, -4, 2, 11, 27, 9]
print(arr)
print(sort(arr))
